src/MiniCover/HitServices/HitService.cs

```diff
@@ -0,0 +1,90 @@
+﻿using MiniCover.HitServices;
+using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Threading;
+
+namespace MiniCover
+{
+    public static class HitService
+    {
+        private static ConcurrentDictionary<string, ConcurrentDictionary<int, Hit>> files = new ConcurrentDictionary<string, ConcurrentDictionary<int, Hit>>();
+
+        public static void Init(string fileName)
+        {
+            files.GetOrAdd(fileName, (f) => new ConcurrentDictionary<int, Hit>());
+        }
+
+        public static MethodContext EnterMethod(string fileName)
+        {
+            var file = files.GetOrAdd(fileName, f => new ConcurrentDictionary<int, Hit>());
+
+            return new MethodContext(file);
+        }
+
+        private static void Save()
+        {
+            var binaryFormatter = new BinaryFormatter();
+
+            foreach (var file in files)
+            {
+                using (var stream = File.Open(file.Key, FileMode.Append, FileAccess.Write, FileShare.None))
+                {
+                    foreach (var kv in file.Value)
+                    {
+                        binaryFormatter.Serialize(stream, kv.Value);
```

Comment:
Adding some serialization because the hit class now is a bit more complex, and contains arrays.
The BinaryFormatter supports appending new data to existing bytes.

Reply:
BinaryFormatter is to slow I think we need to go to protobuff

Reply:
Well, ignore this binaryFormatter, it adds all type information and makes the file bigger.
Maybe we go to another binary serialization later.